packages/testcontainers/src/container-runtime/clients/image/docker-image-client.ts

```diff
@@ -1,38 +1,24 @@
 import Dockerode, { ImageBuildOptions } from "dockerode";
 import byline from "byline";
-import tar from "tar-fs";
-import path from "path";
-import { existsSync, promises as fs } from "fs";
-import dockerIgnore from "@balena/dockerignore";
 import { getAuthConfig } from "../../auth/get-auth-config";
 import { ImageName } from "../../image-name";
 import { ImageClient } from "./image-client";
 import AsyncLock from "async-lock";
 import { log, buildLog, pullLog } from "../../../common";
+import stream from "stream";
 
 export class DockerImageClient implements ImageClient {
   private readonly existingImages = new Set<string>();
   private readonly imageExistsLock = new AsyncLock();
 
   constructor(protected readonly dockerode: Dockerode, protected readonly indexServerAddress: string) {}
 
-  async build(context: string, opts: ImageBuildOptions): Promise<void> {
+  async build(context: stream.Readable, opts: ImageBuildOptions): Promise<void> {
     try {
-      log.debug(`Building image "${opts.t}" with context "${context}"...`);
```

Comment:
What's the reasoning behind moving this logic to the `GenericContainerBuilder`? I like the idea that the `DockerImageClient` can be used standalone and supports `.dockerignore` files out of the box, this is no longer the case

Reply:
I understand your point.

I decided to raise the filtering to a higher abstraction level, recognizing that the `.dockerignore` filtering is a specific instance of passing a tar stream to the container runtime client. This approach streamlines the code and renders the `DockerImageClient` agnostic to the underlying filesystem, enhancing its versatility.

At the time of implementation, I faced two options:

1. Introduce a method in `GenericContainerBuilder` and `DockerImageClient` for passing the stream directly to the container runtime client without `.dockerignore` filtering. This would create an additional flow in both classes.

2. Maintain the `DockerImageClient` agnostic to the filesystem (not utilizing `tar-fs`) and keep the API straightforward by passing a stream to the container runtime client. Since the filtered stream is a subset of the passed stream, this approach avoids needing two separate APIs in both the client and the `GenericContainerBuilder` class.